logrotate.c

```diff
@@ -363,8 +363,33 @@ static int runScript(struct logInfo *log, char *logfn, char *script)
 int createOutputFile(char *fileName, int flags, struct stat *sb, acl_type acl, int force_mode)
 {
     int fd;
-	struct stat sb_create;
-	int acl_set = 0;
+    struct stat sb_create;
+    int acl_set = 0;
+
+    if (stat(fileName, &sb_create) == 0) {
+        /* the destination file already exists, while it should not */
+        if (sb_create.st_size == 0) {
+            /* the file is of zero size, so not needed */
+            if (unlink(fileName) != 0) {
+                message(MESS_ERROR, "error removing already existing output file %s: %s\n",
+                    fileName, strerror(errno));
+                return -1;
+            }
+        } else {
+            struct tm now = *localtime(&nowSecs);
+            size_t fileName_size = sizeof(fileName);
+            char* backupName=alloca(fileName_size+19); /* <fileName>-YYYYMMDDHH.backup */
```

Comment:
Wouldn't this be more clear as `char* backupName = alloca(fileName_size + sizeof("-YYYYMMDDHH.backup"));`?